src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.AppHost/Program.cs

```diff
@@ -0,0 +1,77 @@
+var builder = DistributedApplication.CreateBuilder(args);
+#if (IsOllama) // ASPIRE PARAMETERS
+#elif (IsGHModels)
+
+// You will need to set the endpoint and key to your own values
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+//   dotnet user-secrets set Parameters:gitHubModelsToken YOUR-GITHUB-TOKEN
+var gitHubModelsToken = builder.AddParameter("gitHubModelsToken", secret: true);
+#else // IsAzureOpenAI || IsOpenAI
+
+// You will need to set the connection string to your own value
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+#if (IsOpenAI)
+//   dotnet user-secrets set ConnectionStrings:openai Key=YOUR-API-KEY
+#else // IsAzureOpenAI
+//   dotnet user-secrets set ConnectionStrings:openai Endpoint=https://YOUR-DEPLOYMENT-NAME.openai.azure.com;Key=YOUR-API-KEY
+#endif
+var openai = builder.AddConnectionString("openai");
+#endif
+#if (UseAzureAISearch)
+
+// You will need to set the connection string to your own value
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+//   dotnet user-secrets set ConnectionStrings:azureAISearch Endpoint=https://YOUR-DEPLOYMENT-NAME.search.windows.net;Key=YOUR-API-KEY
+var azureAISearch = builder.AddConnectionString("azureAISearch");
+#elif (UseQdrant)
+
+var qdrantApiKey = builder.AddParameter("qdrantApiKey", secret: true);
+#else // UseLocalVectorStore
+#endif
+#if (IsOllama) // AI SERVICE PROVIDER CONFIGURATION
+
+var ollama = builder.AddOllama("ollama")
+    .WithDataVolume();
+var chat = ollama.AddModel("chat", "llama3.2");
+var embeddings = ollama.AddModel("embeddings", "all-minilm");
+#endif
+#if (UseAzureAISearch) // VECTOR DATABASE CONFIGURATION
+#elif (UseQdrant)
+
+var vectorDB = builder.AddQdrant("vectordb", apiKey: qdrantApiKey, grpcPort: 6334, httpPort: 6333)
+    .WithDataBindMount("./qdrant_data")
+    .WithLifetime(ContainerLifetime.Persistent);
+#else // UseLocalVectorStore
+#endif
+
+var ingestionCache = builder.AddSqlite("ingestionCache")
+    .WithSqliteWeb();
```

Comment:
Could we remove SqliteWeb? We'll be removing the ingestion cache completely soon, so we don't need a web interface for it.

Reply:
Actually it looks as if this isn't persisting the sqlite DB on disk at all right now. Every time I restart the app, it redoes the ingestion completely. Can we update the `AddSqlite` call to pass a path/filename so it persists on disk in a location that survives restarts?

Reply:
I checked and it's sufficient to have:

```cs
var ingestionCache = builder.AddSqlite("ingestionCache", databaseFileName: "ingestioncache.db");
```

Reply:
Actually I was incorrect. If you don't specify a directory, it uses `Path.GetTempPath` and then it ends up sharing the DB across unrelated applications generated from this template.

So it needs to be something like:

```cs
var ingestionCache = builder.AddSqlite("ingestionCache", databasePath: AppContext.BaseDirectory, databaseFileName: "ingestionCache.sqlite");
```

All the more reason for us to remove this when we can :)